### PR TITLE
Allow extra parameters to be sent with the query

### PR DESCRIPTION
**Pull request by DubbleClick**
This should also get rid of all deprecations and clean up the code a bit. It breaks backwards compatibility and is therefore only for a new major version of the bundle.

### CODE REVIEW

**Comment by DubbleClick** [comment]
sorry for the two force pushes, I accidentally rebased onto an old commit

File: DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -39,7 +39,7 @@ public function getConfigTreeBuilder()
                     ->scalarNode('cache')->defaultTrue()->end()
                     ->scalarNode('cache_timeout')->defaultValue(60000)->end()
                     ->scalarNode('width')->defaultNull()->end()
-                    ->scalarNode('object_manager')->defaultValue(1)->end()
+                    ->scalarNode('object_manager')->defaultNull()->end()
```
Comment (DubbleClick) [inline DependencyInjection/Configuration.php]: Why was 1 the default value before? Coming to think of it, why did the object_manager node even exist in the first place?

**Comment by tetranz** [comment]
Thanks for all this @DubbleClick . I only had a quick look but it looks great.

I'll take a closer look, hopefully in the next few days but within the next week at the latest.

There were a couple of minor PRs merged since the last release so I'll tag a final v2.10.1. Then I'll hopefully merge this and I guess I should immediately tag it as v3.0.0.

**Comment by DubbleClick** [comment]
Sure, when you approve of the changes I'll rebase the branch to get rid of some superfluous commits. Also please check out https://github.com/DubbleClick/select2entity-bundle/pull/1 if it's something you'd approve of as default functionality.

**Comment by tetranz** [comment]
@DubbleClick I tested this and it looks good but I found one small issue.

In Tetranz\Select2EntityBundle\Form\Type\Select2EntityType::finishView I had to add `query_parameters` to the array at line 108.

The default Symfony install has `strict_variables: '%kernel.debug%'` in twig.yaml so in debug more Twig will crash if a variable does not exist.
Otherwise it doesn't get through to the view and then `{% if query_parameters %}` crashes.

**Comment by tetranz** [comment]
Sorry, I hit the wrong button, didn't mean to close. :)

I'm ready to merge. I can add that change above but it sounded like you wanted to rebase and tidy up.

File: DependencyInjection/Configuration.php (hunk 1 above)
Comment (tetranz) [reply DependencyInjection/Configuration.php]: Yeah, a default of 1 makes no sense at all.

I think the changes you made are good with the ability to have either a "global" custom manager or per instance.

**Comment by DubbleClick** [comment]
Good catch, I'll rebase now and add it in there.

**Comment by DubbleClick** [comment]
Should be fixed now, you're welcome to merge. :)
Might still add a few features soon and switch to using the bundle again.

**Comment by DubbleClick** [comment]
Edit: Caught some spelling mistakes in the readme. Should be better now.

**Comment by tetranz** [comment]
Thanks. I'll tag this as 3.0.0.

Hopefully not too many people on Symfony 2 or 3 have their composer version constraints set too wide.

I want to tidyup the installation instructions a little. It's a bit outdated now talking about assetic etc. Maybe next weekend.

**Pull request merged**